hack/update-test-codegen.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/library.sh
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 $(dirname $0)/../vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 $(dirname $0)/../vendor/knative.dev/pkg 2>/dev/null || echo ../pkg)}
+
+(
+  cd ${REPO_ROOT_DIR}
+  rm -rf ${REPO_ROOT_DIR}/knative.dev/pkg/apis/test
+)
+
+${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
+  knative.dev/pkg/client/test knative.dev/pkg/apis/test \
+  "example:v1alpha1" \
```

Comment:
You can combine these invocations `"example:v1alpha1 pub:v1alpha1"`

Reply:
In fact, I think not combining them is a bug (unless you output to different test directories)

On that topic, checking the output in will result in our vendoring it into all downstream repos.  I appreciate you adding examples that we can use as tests, but could I bother you a little more to move this so that it's only generates by presubmit (with a `go test` sniff test)?

Reply:
You can't. It puts them in the same client interface and the generators explode from k8s

Reply:
![IMG-20200214-WA0000](https://user-images.githubusercontent.com/32305648/74590966-65e85580-4fc8-11ea-8eaa-8eff56d683ad.jpeg)

Reply:
Yeah, but run in sequence I'd bet that `samplev1alpha1.SampelV1alpha1Client` only contains one of them.  Can we change the groupName of the examples to not so overtly collide?  I care more about the `go test` comment and not vendoring this TBH.

Reply:
I think if you target the same dir twice it will not have the first client in the clientset. I think this is a bug in k8s code generators in general, or working as intended and you should combine the client group into a single dir, or do it like this and target two dirs

Reply:
You can checkout this fork and see what I am talking about. I wanted to do what you are asking but the generators for client did not like it at all

Reply:
I missed that you were targeting separate dirs (sips ☕️ ) can we avoid committing the result but still get test coverage?

Reply:
Another option is to name the second client group like SamplePub with ` // +groupGoName=SamplePub,,`

Reply:
That lets you gen them in the same dir and at the same time without conflict

Reply:
The point of checking in the gen code is that if needs to compile. I bet we could add a gitignore for client/test, that feels a little better...